File: typescript/src/models/order.ts
// this file is @generated

export interface Order {
  id: string;
  checkoutSessionId: string;
  permalinkUrl: string;
}

export const OrderSerializer = {
  _fromJsonObject(object: any): Order {
    return {
      id: object["id"],
      checkoutSessionId: object["checkout_session_id"],
      permalinkUrl: object["permalink_url"],
    };
  },

  _toJsonObject(self: Order): any {
    return {
      id: self.id,
      checkout_session_id: self.checkoutSessionId,
      permalink_url: self.permalinkUrl,
    };
  },
};

File: csharp/Agentic.Commerce.Protocol/Models/Order.cs
// this file is @generated
using System.Text;
using Newtonsoft.Json;

namespace Agentic.Commerce.Protocol
{
    public class Order
    {
        [JsonProperty("id", Required = Required.Always)]
        public required string Id { get; set; }

        [JsonProperty("checkout_session_id", Required = Required.Always)]
        public required string CheckoutSessionId { get; set; }

        [JsonProperty("permalink_url", Required = Required.Always)]
        public required string PermalinkUrl { get; set; }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();

            sb.Append("class Order {\n");
            sb.Append("  Id: ").Append(Id).Append('\n');
            sb.Append("  CheckoutSessionId: ").Append(CheckoutSessionId).Append('\n');
            sb.Append("  PermalinkUrl: ").Append(PermalinkUrl).Append('\n');
            sb.Append("}\n");
            return sb.ToString();
        }
    }
}
